### DISCUSSION TITLE
Multi-page references / bibliography

### DISCUSSION THREAD

**Discussion by pat-s** [quarto-dev/quarto-cli#3149] [Feature Requests]
AFAIU from https://github.com/quarto-dev/quarto-cli/issues/271 and https://quarto.org/docs/presentations/revealjs/index.html#content-overflow, there is currently no way to create multi-page reference slides. 
Currently, if many references are used, these just overflow on the one single slide containing all references.

I'd favor these over the overflowing type, also for creating PDF slides from the HTML output.
Is there any way currently even to achieve this?
If not, it would be great if the item count per page could be defined by the user as the optimal one will depend on the default font size.

**Comment by mcanouil** [comment]
A filter or at worst a post-render script could add the slide separator where it "needs" to be.

**Reply by pat-s** [reply]
Interesting! How would the script locate the correct slide? And apply the insert so often that it does not overflow at the end? 

Anyhow thanks for the idea!

**Reply by mcanouil** [reply]
It would require to targets and walk through some of the HTML div in which the bibliography in embedded.
This can be achieved by looking at the HTML source generated.
Then a Lua filter can be created using a loop and a counter to modify the blocks by adding section (slide) tags.

To give more details I need to take a look at the HTML, but above are the steps I would follow to try to get a multi-slide bibliography.

**Comment by cderv** [comment]
Community question about this: https://community.rstudio.com/t/how-can-i-split-a-quarto-beamer-bibliography-across-two-slides/167682/2

**Comment by jlgraves-ubc** [comment]
I just want to add this is super valuable - I played around a little bit with filters last night trying to get this to work, but it seems like we're really stuck from the user side - it seems that Quarto's writer adds the references after the quarto filters run, so there's no way to interact with the document since the pandoc process we need to adjust is later then we can access.

This is actually really rough for professional presentations, since we routinely need to have a detailed bibliography and it will not fit on a single slide in many cases.

**Reply by jlgraves-ubc** [reply]
A couple of related RMarkdown issues:

* https://github.com/rstudio/rmarkdown/issues/592

**Reply by mcanouil** [reply]
You can change the order of filters:

```yaml
filters:
  - quarto
  - your-filer.lua
```

This "trick" only works for the filters in Quarto, for everything that Quarto does and is not a filter, it won't work.

**Reply by jlgraves-ubc** [reply]
Yeah, I tried that but the `Blocks` object in the `refs` div that quarto produces is empty; I assume its populated using the `citeproc` pandoc filter, but I can't seem to figure out how to execute my filter _after_ that one, even though that order of filters invocation thing seems to indicate that I should be able to; Quarto's filter list doesn't recognize `citeproc`

> `Could not find executable C:\Users\Jonathan Graves\Documents\GitHub\multipart_refs\citeproc`

**Reply by cderv** [reply]
yes for filters that needs to run after citeproc filter, this is a problem. we don't have a post citeproc place in the filter customisation. This will be improve in the future. Follow this one:

* https://github.com/quarto-dev/quarto-cli/issues/4113 

The discussion about citeproc filter place is at 
* https://github.com/quarto-dev/quarto-cli/issues/2294 
 
and there is a workaround possible I believe at https://github.com/quarto-dev/quarto-cli/issues/2294#issuecomment-1238954661 as Pandoc introduced `pandoc.utils.citeproc()` function to run citeproc step from a Lua filter. You need to set `citeproc: false` at YAML level to not have citeproc done twice though. 
Example of filter using this workaround: https://github.com/andrewheiss/quarto-wordcount#how-this-all-works

> This is actually really rough for professional presentations, since we routinely need to have a detailed bibliography and it will not fit on a single slide in many cases.

For now, you can also leverage the `scrollable` class for content overflow in Revealjs presensation. https://quarto.org/docs/presentations/revealjs/index.html#content-overflow 
Not great, but a bit better in some cases. 

Hope it helps.

**Reply by jlgraves-ubc** [reply]
Thanks - much appreciated.  I'll give the workaround a try to see if I can swing it.

**Reply by cestrarz** [reply]
How exactly would separating the bibliography work for a beamer presentation? I am not yet familiar with filters or lua.

**Reply by cestrarz** [reply]
I found a way to do it without either! Setting the class of the frame to {.allowframebreaks} does the trick. Sorry about that, should have looked more before asking.

**Reply by fensch** [reply]
That's perfect! I was searching so long for something like `{.allowframebreaks}`. Thanks @cestrarz !

**Comment by mcanouil** [comment]
Closing in favour of https://github.com/quarto-dev/quarto-cli/issues/4113 and the linked workaround.

**Comment by frodew** [comment]
For anyone stumbling on this issue and looking for an easy solution for a `revealjs/quarto presentation` ("# References {.allowframebreaks}" does not seem to work here):

```
## References {style="font-size: 0.38em;"}

::: {#refs style="column-count: 2; column-gap: 2em;"}
:::
```
